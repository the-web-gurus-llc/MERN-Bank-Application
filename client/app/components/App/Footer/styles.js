import styled from 'styled-components';
import { media } from 'utils';

export const StyledFooter = styled.footer`
  display: inline-block;
  color: black;
  text-align: center;
  position: absolute;
  width: 100%;
  left: 50%;
  transform: translate(-50%, 0);
  bottom: 0;
  padding: 10px;
  border-top: 1px solid rgba(0, 0, 0, 0.12);
  font-size: 12px;
  white-space: nowrap;
  transition: 0.4s;

  ${media.tablet`
    opacity: ${({ collapsed }) => (collapsed ? 0 : 1)}};
    visibility: ${({ collapsed }) => (collapsed ? 'hidden' : 'visible')}};
  `};
`;
