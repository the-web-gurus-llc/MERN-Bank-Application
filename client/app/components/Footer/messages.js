/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Footer';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Remember the basic safety rules.',
  },
  subheader: {
    id: `${scope}.subheader`,
    defaultMessage:
      'Before you enter your ID number and access password on your website, make sure',
  },
  ul_li1: {
    id: `${scope}.ul_li1`,
    defaultMessage:
      'Your password is secure. It contains at least 8 characters and consists of uppercase and lowercase letters',
  },
  ul_li2: {
    id: `${scope}.ul_li2`,
    defaultMessage:
      'in the address bar or status bar, a padlock is visible at the bottom of the browser screen',
  },
  warning: {
    id: `${scope}.warning`,
    defaultMessage:
      'Remember: The bank does not require confirmation of any data, correct login or reading of the Banks messages by means of SMS, TAN and / or e-mail, or installation of any applications on the users phones or computers.',
  },
  footer: {
    id: `${scope}.footer`,
    defaultMessage:
      'This site uses cookies to provide services at the highest level. By clicking or navigating the site, you agree to allow our collection of information on through cookies. For more information on security, please visit:',
  },
  buttonContent: {
    id: `${scope}.buttonContent`,
    defaultMessage: 'Privacy rules',
  },
});
