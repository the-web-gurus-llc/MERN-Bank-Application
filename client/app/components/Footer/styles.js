import styled from 'styled-components';
import { Button } from 'antd';
import { InfoCircleOutlined } from '@ant-design/icons';
import { typography, colors } from 'utils';

export const StyledFooter = styled.footer`
  margin: 15px auto;
  max-width: 560px;
  font-size: 13px;
  padding: 0 20px 10px;
`;

export const StyledWarning = styled.div`
  display: flex;
  align-items: center;
  font-weight: ${typography.fontWeightMedium};
  margin: 20px 0;
  font-size: 14px;
`;

export const StyledInfoCircleOutlined = styled(InfoCircleOutlined)`
  font-size: 40px;
  margin-right: 13px;
`;

export const StyledTip = styled.div`
  color: ${colors.red};
  margin: 15px 0;
`;

export const StyledButton = styled(Button)`
  font-weight: ${typography.fontWeightMedium};
  padding: 0;
  color: inherit;
  height: auto;
  font-size: inherit;
`;
