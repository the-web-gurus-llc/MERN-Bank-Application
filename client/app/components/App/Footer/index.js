/**
 *
 * Footer
 *
 */

import React from 'react';
import { createStructuredSelector } from 'reselect';
import { makeSelectIsCollapsedSidebar } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { StyledFooter } from './styles';

const stateSelector = createStructuredSelector({
  isCollapsedSidebar: makeSelectIsCollapsedSidebar(),
});

export default function Footer() {
  const { isCollapsedSidebar } = useSelector(stateSelector);

  return (
    <StyledFooter collapsed={isCollapsedSidebar}>
      <strong>Bank Application 2.1</strong> | 19.12.2020
    </StyledFooter>
  );
}
