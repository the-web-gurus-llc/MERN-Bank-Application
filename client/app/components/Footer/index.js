/**
 *
 * Footer
 *
 */

import React from 'react';
import { useDispatch } from 'react-redux';
import { push } from 'connected-react-router';
import { FormattedMessage } from 'react-intl';
import { routes } from 'utils';
import {
  StyledFooter,
  StyledButton,
  StyledInfoCircleOutlined,
  StyledWarning,
  StyledTip,
} from './styles';
import messages from './messages';

export default function Footer() {
  const dispatch = useDispatch();

  return (
    <StyledFooter>
      <StyledWarning>
        <StyledInfoCircleOutlined />
        <FormattedMessage {...messages.header} />
      </StyledWarning>

      <div>
        <FormattedMessage {...messages.subheader} />
        <ul>
          <li>
            <FormattedMessage {...messages.ul_li1} />
          </li>
          <li>
            <FormattedMessage {...messages.ul_li2} />
          </li>
        </ul>
      </div>

      <StyledTip>
        <FormattedMessage {...messages.warning} />
      </StyledTip>

      <div>
        <FormattedMessage {...messages.footer} />{' '}
        <StyledButton
          type="link"
          onClick={() => dispatch(push(routes.privacy.path))}
        >
          <FormattedMessage {...messages.buttonContent} />
        </StyledButton>
      </div>
    </StyledFooter>
  );
}
