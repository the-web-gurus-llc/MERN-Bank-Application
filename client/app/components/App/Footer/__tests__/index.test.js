import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import configureStore from 'utils/configureStore';
import { Provider } from 'react-redux';
import Footer from '../index';

describe('<Footer />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
