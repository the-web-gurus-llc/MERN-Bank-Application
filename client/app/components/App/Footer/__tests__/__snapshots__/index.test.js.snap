// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the snapshot 1`] = `
<footer
  class="Footerstyle__StyledFooter-sc-7flfx7-0 iYkHNP"
>
  <strong>
    Bank Application 2.0
  </strong>
   | 10.07.2020
</footer>
`;
